src/bytes.js

```diff
@@ -0,0 +1,1497 @@
+            ret += makehexform(num);
+        } else if (num === 9 || num === 10 || num === 13 || num === 92) {
+            switch (num) {
+                case 9:
+                    ret += "\\t";
+                    break;
+                case 10:
+                    ret += "\\n";
+                    break;
+                case 13:
+                    ret += "\\r";
+                    break;
+                case 92:
+                    ret += "\\\\";
+                    break;
+            }
+        } else {
+            const c = String.fromCharCode(num);
+            ret += c === quote ? "\\" + c : c;
+        }
+    }
+    ret += quote;
+    return new Sk.builtin.str(ret);
+};
+
+Sk.builtin.bytes.prototype.mp$subscript = function (index) {
+    var ret;
+    var i;
+    if (Sk.misceval.isIndex(index)) {
+        i = Sk.misceval.asIndex(index);
+        if (i !== undefined) {
+            if (i < 0) {
+                i = this.v.byteLength + i;
+            }
+            if (i < 0 || i >= this.v.byteLength) {
+                throw new Sk.builtin.IndexError("index out of range");
+            }
+            return new Sk.builtin.int_(this.v[i]);
+        }
+    } else if (index instanceof Sk.builtin.slice) {
+        ret = [];
+        index.sssiter$(this.v.byteLength, (i) => {
+            ret.push(this.v[i]);
+        });
+        return new Sk.builtin.bytes(ret);
+    }
+
+    throw new Sk.builtin.TypeError("byte indices must be integers, not " + Sk.abstr.typeName(index));
+};
+
+Sk.builtin.bytes.prototype.ob$eq = function (other) {
+    if (this === other) {
+        return Sk.builtin.bool.true$;
+    } else if (!(other instanceof Sk.builtin.bytes)) {
+        return Sk.builtin.NotImplemented.NotImplemented$;
+    } else if (this.v.byteLength != other.v.byteLength) {
+        return Sk.builtin.bool.false$;
+    }
+
+    for (let i = 0; i < this.v.byteLength; i++) {
+        if (this.v[i] != other.v[i]) {
+            return Sk.builtin.bool.false$;
+        }
+    }
+
+    return Sk.builtin.bool.true$;
+};
+
+Sk.builtin.bytes.prototype.ob$ne = function (other) {
+    const ret = this.ob$eq(other);
+    if (ret === Sk.builtin.NotImplemented.NotImplemented$) {
+        return ret;
+    }
+    return Sk.misceval.isTrue(ret) ? Sk.builtin.bool.false$ : Sk.builtin.bool.true$;
+};
+
+function slotCompareUint8(f) {
+    return function (other) {
+        if (!(other instanceof Sk.builtin.bytes)) {
+            return Sk.builtin.NotImplemented.NotImplemented$;
+        }
+        return f(this.v, other.v);
+    };
+}
+Sk.builtin.bytes.prototype.ob$lt = slotCompareUint8((v, w) => new Sk.builtin.bool(v < w));
+Sk.builtin.bytes.prototype.ob$le = slotCompareUint8((v, w) => new Sk.builtin.bool(v <= w));
+Sk.builtin.bytes.prototype.ob$gt = slotCompareUint8((v, w) => new Sk.builtin.bool(v > w));
+Sk.builtin.bytes.prototype.ob$ge = slotCompareUint8((v, w) => new Sk.builtin.bool(v >= w));
```

Comment:
These cute comparisons don't work, because they rely on stringification, and if you compare strings then `'10' < '9'`. For a quick and dirty approach you could call `$jsstr()` and compare the results...

Reply:
quick and dirty approach utilized. Can adjust if desired.